src/otk/annotation.py

```diff
@@ -208,6 +205,18 @@ def fspath_with_include(self) -> str:
         return filename
 
 
+if sys.version_info >= (3, 10):
+    class AnnotatedPath(AnnotatedPathBase, pathlib.Path):
+        def __init__(self, *args: Any, **kwargs: Any) -> None:
+            pathlib.Path.__init__(self, *args, **kwargs)
+            AnnotatedPathBase.__init__(self)
+else:
+    class AnnotatedPath(AnnotatedPathBase, pathlib.PosixPath):  # pylint: disable=abstract-method
+        def __init__(self, *args: Any) -> None:  # pylint: disable=W0613
+            pathlib.PosixPath.__init__(self)
+            AnnotatedPathBase.__init__(self)
```

Comment:
Is there a common behaviour we can use here instead of having potentially different behaviour when switching Python versions?  Does using `PosixPath` for both cause issues?
I'm looking at this and thinking it'll be a pain to troubleshoot `AnnotatedPath` behaviour if it does slightly different things based on interpreter version.